### PR TITLE
Displaying TS code examples in the docs

### PR DESCRIPTION
**Pull request by sequba**
### Context
This pull request creates the UI that displays the typescript code examples in the docs and add the ability to switch between JS and TS versions, and add TS to code sandboxes

In the `code` tab, there will be a dropdown with the available versions of the example (JS and TS in most cases). The source code preview area will display the selected version of the code.

- add tsconfig

 convert all jsx to tsx
 - [x] accessibility/
  - [x] cell-features/
  - [x] cell-types/
  - [x] formulas/
  - [x] navigation/
  - [x] rows/
  - [x] tools-and-building/
  - [x] accessories-and-menus/
  - [x] cell-functions/
  - [x] columns/
  - [x] getting-started/
  - [x] internationalization/
  - [x] optimization/
  - [x] security/
  - [x] technical-specification/
  - [x] upgrade-and-migration/

### How has this been tested?
Locally

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/dev-handsontable/issues/1905
2. additionally https://github.com/handsontable/dev-handsontable/issues/1953 is included in this

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:
- [x] Add TypeScript/JavaScript switcher
- [x] Run code in selected language in sandbox

[skip changelog]

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "docs:start:no-cache": "npm run docs:scripts:link-assets && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev --no-cache ./",
     "docs:serve": "http-server -c-1 .vuepress/dist/",
     "docs:build": "npm run docs:scripts:generate-js-examples && cross-env NODE_OPTIONS=--openssl-legacy-provider node .vuepress/tools/build.mjs -- --no-cache",
```
Comment (sequba) [inline docs/package.json:21]:
```suggestion
    "docs:build": "npm run docs:scripts:generate-js-examples && npm run docs:build:no-examples",
```

File: docs/package.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
...
+    "docs:build:no-examples ": "cross-env NODE_OPTIONS=--openssl-legacy-provider node .vuepress/tools/build.mjs -- --no-cache",
```
Comment (sequba) [inline docs/package.json:22]:
```suggestion
    "docs:build:no-examples": "cross-env NODE_OPTIONS=--openssl-legacy-provider node .vuepress/tools/build.mjs -- --no-cache",
```

File: docs/content/guides/cell-types/handsontable-cell-type/javascript/example1.ts [hunk 3]
```diff
@@ -61,5 +61,6 @@ new Handsontable(container, {
     },
   ],
   autoWrapRow: true,
-  autoWrapCol: true
+  autoWrapCol: true,
+  height: 'auto',
```
Comment (sequba) [inline docs/content/guides/cell-types/handsontable-cell-type/javascript/example1.ts:65]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/dropdown-cell-type/react/example1.jsx [hunk 4]
```diff
@@ -29,6 +29,7 @@ const ExampleComponent = () => {
       ]}
       autoWrapRow={true}
       autoWrapCol={true}
+      height="auto"
```
Comment (sequba) [inline docs/content/guides/cell-types/dropdown-cell-type/react/example1.jsx:32]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/dropdown-cell-type/javascript/example1.ts [hunk 5]
```diff
@@ -25,5 +25,6 @@ new Handsontable(container, {
   ],
   autoWrapRow: true,
   autoWrapCol: true,
+  height: 'auto',
```
Comment (sequba) [inline docs/content/guides/cell-types/dropdown-cell-type/javascript/example1.ts:28]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/autocomplete-cell-type/react/example3.jsx [hunk 6]
```diff
@@ -31,6 +31,7 @@ const ExampleComponent = () => {
         {}, // Chassis color is a default text column
         {} // Bumper color is a default text column
       ]}
+      height="auto"
```
Comment (sequba) [inline docs/content/guides/cell-types/autocomplete-cell-type/react/example3.jsx:34]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/autocomplete-cell-type/react/example2.jsx [hunk 7]
```diff
@@ -44,6 +44,7 @@ const ExampleComponent = () => {
           allowInvalid: true //true is default
         }
       ]}
+      height="auto"
```
Comment (sequba) [inline docs/content/guides/cell-types/autocomplete-cell-type/react/example2.jsx:47]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/autocomplete-cell-type/react/example1.jsx [hunk 8]
```diff
@@ -41,6 +41,7 @@ const ExampleComponent = () => {
           trimDropdown: false
         }
       ]}
+      height="auto"
```
Comment (sequba) [inline docs/content/guides/cell-types/autocomplete-cell-type/react/example1.jsx:44]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/autocomplete-cell-type/javascript/example3.ts [hunk 9]
```diff
@@ -27,5 +27,6 @@ new Handsontable(container, {
     {} // Bumper color is a default text column
   ],
   autoWrapRow: true,
-  autoWrapCol: true
+  autoWrapCol: true,
+  height: 'auto'
```
Comment (sequba) [inline docs/content/guides/cell-types/autocomplete-cell-type/javascript/example3.ts:31]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/autocomplete-cell-type/javascript/example2.ts [hunk 10]
```diff
@@ -39,5 +39,6 @@ new Handsontable(container, {
     }
   ],
   autoWrapRow: true,
-  autoWrapCol: true
+  autoWrapCol: true,
+  height: 'auto'
```
Comment (sequba) [inline docs/content/guides/cell-types/autocomplete-cell-type/javascript/example2.ts:43]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/autocomplete-cell-type/javascript/example1.ts [hunk 11]
```diff
@@ -36,5 +36,6 @@ new Handsontable(container, {
     }
   ],
   autoWrapRow: true,
-  autoWrapCol: true
+  autoWrapCol: true,
+  height: 'auto'
```
Comment (sequba) [inline docs/content/guides/cell-types/autocomplete-cell-type/javascript/example1.ts:40]: Is this change done on purpose or is it a merge glitch?

File: docs/content/guides/cell-types/autocomplete-cell-type/autocomplete-cell-type.md [hunk 12]
```diff
@@ -115,9 +120,11 @@ Autocomplete can also be used with Ajax data sources. In the example below, sugg
 
 ### Related guides
 
+<div class="boxes-list gray">
 - [Cell type](@/guides/cell-types/cell-type/cell-type.md)
 - [Dropdown cell type](@/guides/cell-types/dropdown-cell-type/dropdown-cell-type.md)
 - [Select cell type](@/guides/cell-types/select-cell-type/select-cell-type.md)
+</div>
```
Comment (sequba) [inline docs/content/guides/cell-types/autocomplete-cell-type/autocomplete-cell-type.md:127]: Is this change done on purpose or is it a merge glitch?

File: docs/README.md [hunk 13]
```diff
@@ -48,6 +48,8 @@ From the `docs` directory, you can run the following npm scripts:
 * `npm run docs:scripts:link-assets` – Prepares the `next` documentation version's CSS and JavaScript.
 * `npm run docs:review [COMMIT_HASH]` – Deploys the documentation locally at a `[COMMIT_HASH]` commit.
 * `npm run docs:test:example-checker` – Runs the tests that checks if all Docs examples work.
+* `npm run docs:scripts:generate-js-examples` – Generate js examples from ts files (need to run this script before first `docs:start` and every time after change ts file content in `docs/content/guides`)
```
Comment (sequba) [inline docs/README.md:51]:
```suggestion
* `npm run docs:scripts:generate-js-examples` – Generate js version of the code examples (need to run this script before first `docs:start` and every time some `*.ts` inside `docs/content/guides` changes)
```

Comment (sequba) [inline docs/README.md:51]: Please, add `docs:scripts:generate-js-examples` as a step in `Getting started with Handsontable documentation` section of the readme.

**Review by sequba — COMMENTED** [review]
@adrianspdev please look at my comments. Some of the changes seem to be unrelated to TS code examples. Please, double check, if they should be on this branch.

File: docs/README.md (hunk 13 above)
Comment (adrianspdev) [reply docs/README.md]: added

**Comment by magierg** [comment]
some react demos will not work in codesandbox ie. https://dev.handsontable.com/docs/react-data-grid/binding-to-data/

**Comment by magierg** [comment]
Angular, Vue2, Vue3 demos are broken when opening in CS, Stackblitz or JS Fiddle

**Review by adrianspdev — APPROVED** [review]

**Review by evanSe — APPROVED** [review]

**Pull request merged**